docs/password_hashing.rst

```diff
@@ -111,23 +160,71 @@ the decryption would fail and an exception would be raised::
 
     >>> from nacl import pwhash, secret, utils
     >>>
-    >>> ops = pwhash.SCRYPT_OPSLIMIT_SENSITIVE
-    >>> mem = pwhash.SCRYPT_MEMLIMIT_SENSITIVE
+    >>> ops = pwhash.ARGON2I_OPSLIMIT_SENSITIVE
+    >>> mem = pwhash.ARGON2I_MEMLIMIT_SENSITIVE
     >>>
-    >>> salt = utils.random(pwhash.SCRYPT_SALTBYTES)
+    >>> salt = utils.random(pwhash.ARGON2I_SALTBYTES)
     >>>
     >>> guessed_pw = b'I think Alice shared this password with Bob'
     >>>
-    >>> Eves_key = pwhash.kdf_scryptsalsa208sha256(secret.SecretBox.KEY_SIZE,
-    ...                                            guessed_pw, salt,
-    ...                                            opslimit=ops, memlimit=mem)
+    >>> Eves_key = pwhash.kdf_argon2i(secret.SecretBox.KEY_SIZE,
+    ...                               guessed_pw, salt,
+    ...                               opslimit=ops, memlimit=mem)
     >>> Eves_box = secret.SecretBox(Eves_key)
     >>> intercepted = Eves_box.decrypt(encrypted)
     Traceback (most recent call last):
         ...
     nacl.exceptions.CryptoError: Decryption failed. Ciphertext failed ...
 
+
+Contrary to the hashed password storage case, where a serialization
```

Comment:
Small rephrasing:

Contrary to the hashed password storage case where a serialization format is well-defined, in the raw key derivation case the library user must take care to store (and retrieve) both a reference to the kdf used to derive the secret key and all the derivation parameters. These parameters are needed to later generate the same secret key from the password.